crates/sui-core/src/consensus_handler.rs

```diff
@@ -1360,17 +1376,18 @@ impl ConsensusBlockHandler {
                         .inc();
                     continue;
                 }
-                self.epoch_store
-                    .set_consensus_tx_status(position, ConsensusTxStatus::FastpathCertified);
-
                 self.metrics
                     .consensus_block_handler_txn_processed
                     .with_label_values(&["certified"])
                     .inc();
+
                 if let ConsensusTransactionKind::UserTransaction(tx) = parsed.transaction.kind {
                     if tx.is_consensus_tx() {
                         continue;
                     }
+                    // Only set fastpath certified status on transactions intended for fastpath execution.
+                    self.epoch_store
+                        .set_consensus_tx_status(position, ConsensusTxStatus::FastpathCertified);
```

Comment:
IIUC this shouldn't have any material change to the system behaviour since on the `wait_for_effects` will always wait for execution before it returns any result , right?